server/asset/eth/coiner.go

```diff
@@ -145,17 +146,23 @@ func (eth *Backend) baseCoin(coinID []byte, contractData []byte) (*baseCoin, err
 	// transaction with a very low gas price may need to be resent with a
 	// higher price.
 	zero := new(big.Int)
-	rate := tx.GasPrice()
-	if rate == nil || rate.Cmp(zero) <= 0 {
-		rate = tx.GasFeeCap()
-		if rate == nil || rate.Cmp(zero) <= 0 {
-			return nil, fmt.Errorf("Failed to parse gas price from tx %s", txHash)
-		}
+
+	gasFeeCap := tx.GasFeeCap()
+	if gasFeeCap == nil || gasFeeCap.Cmp(zero) <= 0 {
+		return nil, fmt.Errorf("Failed to parse gas fee cap from tx %s", txHash)
+	}
+	gasFeeCapGwei, err := dexeth.WeiToGweiUint64(gasFeeCap)
+	if err != nil {
+		return nil, fmt.Errorf("unable to convert gas fee cap: %v", err)
 	}
 
-	gasPrice, err := dexeth.WeiToGweiUint64(rate)
+	gasTipCap := tx.GasTipCap()
+	if gasTipCap == nil || gasTipCap.Cmp(zero) <= 0 {
```

Comment:
With a legacy transaction, which returns the same gas price for both `GasFeeCap` and `GasTipCap`, I think that the `swapCoin.gasTipCap` check in `(*Backend).ValidateFeeRate` is wrong or at least misleading.
Say we are debugging and we inspect a `swapCoin`, we'd see a `gasTipCap == gasFeeCap` (a huge `gasTipCap`).
I guess we can solve this with documentation spelling out the logic of why a legacy txn can be treated as a dynamic tx with a massive tip, but I do think we should spell that out in both methods (`baseCoin` construction and `ValidateFeeRate`)

Reply:
I added some docs.